deeprank/features/atomic_contacts.py

```diff
@@ -0,0 +1,565 @@
+from deeprank.models.patch import PatchActionType
+
+_log = logging.getLogger(__name__)
+
+
+class ResidueSynonymCriteria:
+    """The ResidueSynonymCriteria is an object that holds the criteria
+       for a residue to have a certain synonym. It does not hold the synonym string itself however.
+    """
+
+    def __init__(self, residue_name, atoms_present, atoms_absent):
+        """Build new criteria
+
+        Args:
+            residue_name (string): the name of the residue
+            atoms_present (list of strings): the names of the atoms that should be present in the residue
+            atoms_absent (list of strings) the names of the atoms that should be absent in the residue
+        """
+
+        self.residue_name = residue_name
+        self.atoms_present = atoms_present
+        self.atoms_absent = atoms_absent
+
+    def matches(self, residue_name, atom_names):
+        """Check whether the given residue matches this set of criteria
+
+        Args:
+            residue_name (string): the name of the residue to match
+            atom_names (list of strings): the names of the atoms in the residue
+        """
+
+        if self.residue_name != 'all' and residue_name != self.residue_name:
+            return False
+
+        for atom_name in self.atoms_present:
+            if atom_name not in atom_names:
+                return False
+
+        for atom_name in self.atoms_absent:
+            if atom_name in atom_names:
+                return False
+
+        return True
+
+
+def get_squared_distance(pos1, pos2):
+    """Get the squared distance between two positions.
+       This is less computationally expensive than the normal distance
+       and should be used if one does not necessarily need
+       the normal distance for a large set of coordinates.
+
+    Args:
+        pos1 (array of 3): the xyz coords of position 1
+        pos2 (array of 3): the xyz coords of position 2
+    """
+
+    return numpy.sum(numpy.square(pos1 - pos2))
+
+def get_distance(pos1, pos2):
+    """Get the distance between two positions.
+       This is more computationally expensive than the squared distance
+       and should only be used when one really needs this distance.
+
+    Args:
+        pos1 (array of 3): the xyz coords of position 1
+        pos2 (array of 3): the xyz coords of position 2
+    """
+
+    return numpy.sqrt(get_squared_distance(pos1, pos2))
+
+def wrap_values_in_lists(dict_):
+    """Wrap the dictionary's values in lists. This
+       appears to be necessary for the exported features to work.
+
+    Args:
+        dict_(dictionary): the dictionary that should be converted
+    """
+
+    return {key: [value] for key,value in dict_.items()}
+
+class _PhysicsStorage:
+    "A helper object that holds the physics values while summing them"
+
+    ATOM_KEY = ["chainID", "resSeq", "resName", "name"]
+
+    EPSILON0 = 1.0
+    COULOMB_CONSTANT = 332.0636
+
+    VANDERWAALS_DISTANCE_OFF = 8.5
+    VANDERWAALS_DISTANCE_ON = 6.5
```

Comment:
van der Waals forces tend towards zero beyond 4.5 Angstrom.
So may be we can change 6.5 to 4.5?